### PR TITLE
refactor(tx) Switch `EnableAmendment` to new transaction scheme

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

Moves `EnableAmendment` to the new transaction schema, along with hooks, typescript, and more tests.

Additionally moves utility functions out of the Simple component to `amendmentUtils.ts` so that it can be used for #210.

### Context of Change

Completes #206

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

### CODE REVIEW

File: src/containers/shared/components/Transaction/EnableAmendment/Simple.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    case 65536:
+      amendmentStatus = 'Got Majority'
+      expectedDate = getExpectedDate(data.instructions.date, language)
+      break
+    case 131072:
+      amendmentStatus = 'Lost Majority'
+      break
+    default:
+      amendmentStatus = states.unknown
+  }
```
Comment (mvadari) [inline src/containers/shared/components/Transaction/EnableAmendment/Simple.tsx:65]: Shouldn't this be in a `useEffect`?

**Review by jonathanlei — APPROVED** [review]
LGTM other than @mvadari's comment on combining regex

File: src/containers/shared/amendmentUtils.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  const amendmentNames: string[] = []
+  text.split('\n').forEach((line: string) => {
+    const name = line.match(/^\s*REGISTER_F[A-Z]+\s*\((\S+),\s*.*$/)
+    if (name) {
+      amendmentNames.push(name[1])
+    }
+    const name2 = line.match(/^ .*retireFeature\("(\S+)"\)[,;].*$/)
+    if (name2) {
+      amendmentNames.push(name2[1])
+    }
```
[the remark this answers is not in the document]
Comment (ckniffen) [reply src/containers/shared/amendmentUtils.ts:45]: Feature.h does not contain all of the amendments.

[the remark this answers is not in the document]
Comment (ckniffen) [reply src/containers/shared/amendmentUtils.ts:45]: Escrow for example is missing.

[the remark this answers is not in the document]
Comment (mvadari) [reply src/containers/shared/amendmentUtils.ts:45]: Perhaps they could still be combined into one regex in Feature.cpp

**Review by mvadari — APPROVED** [review]

**Pull request merged**